src/transport/emitter.ts

```diff
@@ -91,16 +92,31 @@ export function emitterFor(fn: TransportFunction, options = { binding: HTTP, mod
  *
  * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md
  * @see https://github.com/cloudevents/spec/blob/v1.0/http-protocol-binding.md#13-content-modes
- * @deprecated Will be removed in 4.0.0. Consider using the emitterFactory
  *
  */
-export class Emitter {
+export class Emitter extends EventEmitter {
+  /**
+   * Singleton store
+   */
+  static instance: Emitter | undefined = undefined;
   url?: string;
   protocol: Protocol;
   binaryEmitter: EmitterFunction;
   structuredEmitter: EmitterFunction;
 
+  /**
+   * Create an Emitter
+   * On v4.0.0 this class will only remains as Singleton to allow using the
+   * EventEmitter of NodeJS
+   *
+   *
+   * @param {Object} [options] The configuration options for this event. Options
+   * provided will be passed along to Node.js `http.request()`.
+   * https://nodejs.org/api/http.html#http_http_request_options_callback
+   * @deprecated Will be removed in 4.0.0. Consider using the emitterFactory
```

Comment:
If the `Emitter` class will remain and extend `EventEmitter`, the constructor can't really be removed, can it? Maybe the text should be something more like this?

 ```suggestion
   * @deprecated in 4.0.0 this class functionality will be limited to events propagated throughout the Node.js process. To emit events over a transport protocol, consider using the emitterFactory
```